play-json/shared/src/main/scala/JsResult.scala

```diff
@@ -197,8 +197,33 @@ sealed trait JsResult[+A] { self =>
 }
 
 object JsResult {
+  import scala.util.{ Failure, Try, Success }
   import play.api.libs.functional._
 
+  case class Exception(cause: JsError)
+    extends IllegalArgumentException(Json stringify JsError.toJson(cause))
```

Comment:
Normally you'd use `IllegalArgumentException` to indicate that input arguments are malformed in some way. It generally indicates a programmer error. I've never seen it used when validating user input and I don't think a typical Scala user would expect that either.